### PR TITLE
Feature/add lockable core features

### PR DESCRIPTION
**Pull request by dylanjustice**
Adding the aspect of "Locking" resources to the core library.

There is a need across projects to prevent users from modifying the same record(s) simultaneously. This provides the interfaces and the base class for the implementation of classes implementing the Lockable interface.

### CODE REVIEW

**Comment by KevinBusch** [comment]
Nice!  This looks like it was taken from what we did to Edio with it's locking mechanism?

**Comment by dylanjustice** [comment]
Adding coverage to the model computed properties

**Comment by dylanjustice** [comment]
> Nice! This looks like it was borrowed from what we did to the LMS with it's locking mechanism?

@KevinBusch Yessir.

File: test/AndcultureCode.CSharp.Core.Tests/Stubs/LockableEntity.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+using AndcultureCode.CSharp.Core.Models;
+
+namespace AndcultureCode.CSharp.Core.Tests.Stubs
+{
+    public class LockableEntity: Lockable
+    {
+        public string Name { get; set; }
+
```
Comment (brandongregoryscott) [inline test/AndcultureCode.CSharp.Core.Tests/Stubs/LockableEntity.cs:8]: Minor, extra new line here

File: test/AndcultureCode.CSharp.Core.Tests/Models/Entities/LocakbleTest.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        {
+            // Arrange
+            var sut = new LockableEntity {LockedUntil = DateTimeOffset.Now.AddMinutes(-15)};
+
+            // Act & Assert
+            sut.IsLocked.ShouldBeFalse();
+        }
+
+        #endregion IsLocked
+
```
Comment (brandongregoryscott) [inline test/AndcultureCode.CSharp.Core.Tests/Models/Entities/LocakbleTest.cs:53]: Minor, extra new line here

**Comment by brandongregoryscott** [comment]
Looks good! Thanks for pulling this in.

**Comment by wintondeshong** [comment]
@all-contributors please add @dylanjustice for code and tests

**Pull request merged**